crates/ruff_python_parser/src/parser/mod.rs

```diff
@@ -108,16 +111,11 @@ impl<'src> Parser<'src> {
             tokens,
             recovery_context: RecoveryContext::empty(),
             prev_token_end: TextSize::new(0),
```

Comment:
Should this be set to `start_offset` for the `if self.last_token_end <= start` check in `node_range` to work as expected?